public/js/offline.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2015-2016, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-env browser */
+
+export default class Offline {
+  /**
+   * All elements with class .gulliver-online-aware will:
+   * have an 'online' dataset property that reflects the current online state.
+   * receive a 'change' event whenever the state changes.
+   */
+  static setupEventhandlers() {
+    const body = document.querySelector('body');
+    window.addEventListener('online', () => {
```

Comment:
Move the event listener into an instance method and assign `const onlineAware = document.querySelectorAll('.gulliver-online-aware');` to a field -> no longer needs to be queried every time the status changes. Alternative is to initalize them lazy via internal getter method.